borg/hashindex.pyx

```diff
@@ -212,7 +221,10 @@ cdef class ChunkIndex(IndexBase):
             raise KeyError(key)
         cdef int32_t refcount = _le32toh(data[0])
         if refcount != self.max_ref:
-            refcount -= 1
+            if self.max_ref - refcount < 1:
```

Comment:
Good point. max_ref isn't a cdef and as such is a Python int object (~unlimited resolution). Python's RichCompare (which is also what is used for comparing stuff in Python if you write `><==` etc.) is emitted for this, together with an `int` instanciation from int32_t. So that's at least one round-trip to the allocator, we don't want that.

I'll change it to a cdef, so we get C semantics and no intermediary objects. The test will want a getter for that.